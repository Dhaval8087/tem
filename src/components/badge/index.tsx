import './badge.scss';

import React, { memo } from 'react';

export type IBadgeType = 'badge-success' | 'badge-info' | 'badge-warning' | 'badge-danger'
export interface IBadgeProp {
    text: string,
    isPill?: boolean,
    type: IBadgeType
}

export const Badge = memo(({ type, isPill = true, text }: IBadgeProp) => (
    <span className={`badge ${type} ${!isPill ? '' : 'badge-pill'}`}>{text}</span>
))