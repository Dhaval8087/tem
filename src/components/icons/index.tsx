import React, { memo } from 'react';

export interface IIconProp {
    iconName: string,
    pathCount: number,
}
const loadIcon = (pathCount: number) => {
    const icon = [];
    for (let i = 1; i < pathCount; i++) {
        icon.push(<i key={`path${i}`} className={`path${i}`}></i>)
    }
    return icon;
}
export const RenderIcon = memo(({ iconName, pathCount }: IIconProp) => (
    <i className={iconName}>
        {loadIcon(pathCount)}
    </i>
))