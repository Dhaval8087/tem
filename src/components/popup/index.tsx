import React from 'react';

import { addPopUp, removePopup } from '../../utils/common-helper';

export interface IPopUpProp {
    title: string,
    body?: React.ReactNode,
    primaryText?: string,
    secondaryText?: string,
    onPrimaryHandler(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void,
    onSecondaryHandler(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void,
    show: boolean,
    id?: string
}

export const Popup = ({
    title,
    body,
    primaryText = 'Save',
    secondaryText = 'Cancel',
    onPrimaryHandler,
    onSecondaryHandler,
    show,
    ...rest
}: IPopUpProp) => {
    if (!show) {
        removePopup();
        return null;
    }
    addPopUp();
    return (
        <div className="modal fade show" style={{ display: 'block' }} role="dialog" tabIndex={-1} >
            <div className="modal-dialog" role="document">
                <div className="modal-content">
                    <div className="modal-header">
                        <h5 className="modal-title">{title}</h5>
                        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                        </button>
                    </div>
                    <div className="modal-body">
                        {body}
                    </div>
                    <div className="modal-footer">
                        <button type="button" className="btn btn-primary" id={rest.id} onClick={onPrimaryHandler}>{primaryText}</button>
                        <button type="button" className="btn btn-secondary" data-dismiss="modal" onClick={onSecondaryHandler}>{secondaryText}</button>
                    </div>
                </div>
            </div>
        </div>
    )
}