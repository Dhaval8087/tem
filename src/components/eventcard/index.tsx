import './eventcard.scss';

import React, { FC } from 'react';

import { EventModel, EventType } from '../../models';
import { eventDuration, formatDate, getOneCookie } from '../../utils';
import { Badge } from '../badge';
import { RenderIcon } from '../icons';

export interface IProps {
    event: EventModel,
    eventSubscriptionHandler(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void,
    subscriptionText: string,
    eventType: EventType
}

export const EventCard: FC<IProps> = ({ event, eventSubscriptionHandler, subscriptionText, eventType }) => {
    const startDate = new Date(event.startDate);
    const endDate = new Date(event.endDate);
    const selectedEvents = (getOneCookie('selectedevents') as unknown) as Array<EventModel | undefined>;
    let isSignup = true;

    if (eventType === 'ALL' && selectedEvents && selectedEvents.find(p => p?.id === event.id)) {
        isSignup = false;
    }
    return (
        <div className="card">
            <div className="card-block">
                <div className="row m-2">
                    <div className="col-md-10 inline-box">
                        {event.isFree && <Badge type="badge-success" text="FREE"  />}
                        <h6 className="ml-3 font-weight-bold">{event.name}</h6>
                    </div>
                    <div className="col-4 col-md-2 btn-signup">
                        {isSignup ? <button type="button" className="btn btn-outline-secondary" id={event.id.toString()} onClick={eventSubscriptionHandler}><b>{subscriptionText}</b></button> : null}
                    </div>
                </div>
                <div className="row m-2">
                    <div className="col-md-4 inline-box">
                        <i className="icon-world"></i>
                        <span className="text-muted city">{event.city}</span>
                    </div>
                    <div className="col-md-4 inline-box">
                        <RenderIcon pathCount={8} iconName="icon-hourglass" />
                        <span className="text-muted">{`${eventDuration(startDate, endDate)}`}'</span>
                    </div>
                    <div className="col-md-4 text-right ">
                        <div className="inline-box">
                            <RenderIcon pathCount={22} iconName="icon-clock" />
                            <span className="text-muted">{` from ${formatDate(startDate)} to ${formatDate(endDate)}`}</span>
                        </div>

                    </div>
                </div>
            </div>
        </div>
    );
}

