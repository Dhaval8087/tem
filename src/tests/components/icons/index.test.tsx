import { shallow } from 'enzyme';
import React from 'react';

import { IIconProp, RenderIcon } from '../../../components/icons';

function renderIcone(args?: {}) {
    const defaultProps: IIconProp = {
        iconName: 'icon-hourglass',
        pathCount: 22,
    };
    const props = { ...defaultProps, ...args }
    return shallow(<RenderIcon {...props} />);
}
it("render badge", () => {
    const wrapper = renderIcone();
    expect(wrapper).toMatchSnapshot();
})
it("Should generate the proper path", () => {
    const wrapper = renderIcone();
    expect(wrapper.find('i').length).toBe(22);
})
it("Should generate the root span with proper icon name",()=>{
    const wrapper = renderIcone({iconName:'icon-clock',pathCount:8});
    const firstElement = wrapper.find("i").first();
    expect(firstElement.hasClass('icon-clock')).toBe(true);
})

