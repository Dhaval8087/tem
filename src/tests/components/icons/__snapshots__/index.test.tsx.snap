// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render badge 1`] = `
<i
  className="icon-hourglass"
>
  <i
    className="path1"
    key="path1"
  />
  <i
    className="path2"
    key="path2"
  />
  <i
    className="path3"
    key="path3"
  />
  <i
    className="path4"
    key="path4"
  />
  <i
    className="path5"
    key="path5"
  />
  <i
    className="path6"
    key="path6"
  />
  <i
    className="path7"
    key="path7"
  />
  <i
    className="path8"
    key="path8"
  />
  <i
    className="path9"
    key="path9"
  />
  <i
    className="path10"
    key="path10"
  />
  <i
    className="path11"
    key="path11"
  />
  <i
    className="path12"
    key="path12"
  />
  <i
    className="path13"
    key="path13"
  />
  <i
    className="path14"
    key="path14"
  />
  <i
    className="path15"
    key="path15"
  />
  <i
    className="path16"
    key="path16"
  />
  <i
    className="path17"
    key="path17"
  />
  <i
    className="path18"
    key="path18"
  />
  <i
    className="path19"
    key="path19"
  />
  <i
    className="path20"
    key="path20"
  />
  <i
    className="path21"
    key="path21"
  />
</i>
`;
